src/disk.cc

```diff
@@ -136,125 +137,150 @@ struct lkl_dev_blk_ops lkl_js_disk_ops = {
-			disk.handle = js_ops;
-		}
-
-		~MountWorker() {}
-
-		void Execute () {
-			disk_id = lkl_disk_add(&disk);
-			ret = lkl_mount_dev(
-					disk_id,
-					part,
-					fs_type,
-					readonly ? LKL_MS_RDONLY : 0, NULL,
-					mountpoint,
-					sizeof(mountpoint)
-			);
-		}
-
-		void HandleOKCallback () {
-			HandleScope scope;
-
-			if (ret < 0) {
-				v8::Local<v8::Value> argv[] = {
-					ErrnoException(-ret)
-				};
-				callback->Call(1, argv);
-			} else {
-				v8::Local<v8::Object> ret2 = New<v8::Object>();
-				Set(ret2, New("mountpoint").ToLocalChecked(), New(mountpoint).ToLocalChecked());
-				Set(ret2, New("diskId").ToLocalChecked(), New(disk_id));
-				v8::Local<v8::Value> argv[] = { Null(), ret2 };
-				callback->Call(2, argv);
-			}
-		}
-
-	private:
-		long ret;
-		bool readonly;
-		unsigned int part;
-		char fs_type[10];
-		char mountpoint[32];
-		unsigned int disk_id;
-		lkl_disk_t disk;
+struct mount_state_t {
+	long ret;
+	unsigned int disk_id;
+	char* mountpoint;
+	Callback *callback;
 };
 
+static void mount_callback(mount_state_t *s) {
+	HandleScope scope;
+	if (s->ret < 0) {
+		v8::Local<v8::Value> argv[] = {ErrnoException(-s->ret)};
+		s->callback->Call(1, argv);
+	} else {
+		v8::Local<v8::Object> ret2 = New<v8::Object>();
+		Set(
+			ret2,
+			New("mountpoint").ToLocalChecked(),
+			New(s->mountpoint).ToLocalChecked()
+		);
+		Set(ret2, New("diskId").ToLocalChecked(), New(s->disk_id));
+		v8::Local<v8::Value> argv[] = { Null(), ret2 };
+		s->callback->Call(2, argv);
+	}
+}
+
+struct mount_args_t {
+	lkl_disk_t disk;
+	bool readonly;
+	const char* fs_type;
+	unsigned int part;
+	Callback *callback;
+};
+
+void do_mount(mount_args_t* args) {
+	char mountpoint[32];
+	unsigned int disk_id = lkl_disk_add(&(args->disk));
+	long ret = lkl_mount_dev(
+		disk_id,
+		args->part,
+		args->fs_type,
+		args->readonly ? LKL_MS_RDONLY : 0, NULL,
+		mountpoint,
+		sizeof(mountpoint)
+	);
+	mount_state_t s = {ret, disk_id, mountpoint, args->callback};
+	run_on_default_loop(mount_callback, &s);
+	delete args->fs_type;
+	delete args->callback;
+	delete args;
+}
+
+void mount_entry(NAN_METHOD_ARGS_TYPE info) {
+	bool readonly = info[0]->BooleanValue();
+
+	char *fs_type = new char[10];
+	Utf8String fs_type_(info[1]);
+	unsigned int part = info[2]->Uint32Value();
+	strncpy(fs_type, *fs_type_, sizeof(fs_type) - 1);
+	fs_type[sizeof(fs_type) - 1] = '\0';
+
+	// FIXME: delete this object when the disk is unmounted
+	auto js_ops = new js_ops_t;
+	js_ops->request = new Callback(info[3].As<v8::Function>());
+	js_ops->get_capacity = new Callback(info[4].As<v8::Function>());
+
+	Callback *callback = new Callback(info[5].As<v8::Function>());
+
+	lkl_disk_t disk;
+	disk.ops = &lkl_js_disk_ops;
+	disk.handle = js_ops;
+
+	mount_args_t* args = new mount_args_t;
+	args->disk = disk;
+	args->readonly = readonly;
+	args->fs_type = fs_type;
+	args->part = part;
+	args->callback = callback;
+
+	run_on_worker_loop(do_mount, args);
```

Comment:
instead of implementing our own threadpool and work scheduling routines, can we create a separate uv_pool and then use the normal [uv_queue_work](http://docs.libuv.org/en/v1.x/threadpool.html#c.uv_queue_work) and  `uv_async`  APIs to do queue work?